metabase/metabase
# Add warning message when leaving an edited, existing SQL question

### Description

Adds "Are you sure?" prompt when a user tries to refresh/close/navigate away from the current page, after the user has edited an existing SQL question.


### How to verify

1. Go to an existing SQL question
2. Edit the query (ex: click `Open Editor` and add/delete a character in the SQL query)
3. Try to refresh/close tab/go to a different URL, you should see a message similar to that of the video below.

### Demo

(I try to ⌘-W from the tab)

https://user-images.githubusercontent.com/25306947/235925154-1b964312-e040-45ad-8296-7ab6d9e3f44c.mov


### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30477)
